packages/docusaurus/src/webpack/utils.ts

```diff
@@ -148,3 +156,53 @@ export function applyConfigureWebpack(
   }
   return config;
 }
+
+export async function choosePort(
```

Comment:
what about putting everything in a choosePort.ts file? I'd like to have all this feature in a single place (unlike CRA), and maybe comment that this code is inspired from CRA codebase would be useful